src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,87 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String CSV_DELIMITER = ",";
+    private static final String OPERATION_SUPPLY = "supply";
+    private static final String OPERATION_BUY = "buy";
+    private static final String REPORT_SUPPLY = "supply,";
+    private static final String REPORT_BUY = "buy,";
+    private static final String REPORT_RESULT = "result,";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+
     public void getStatistic(String fromFileName, String toFileName) {
+        Totals totals = readAndAggregate(fromFileName);
+        String report = buildReport(totals.supply, totals.buy);
+        writeReport(toFileName, report);
+    }
+
+    private Totals readAndAggregate(String fromFileName) {
+        int totalBuy = 0;
+        int totalSupply = 0;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                try {
+                    String[] parts = line.split(CSV_DELIMITER);
+                    if (parts.length != 2) {
+                        throw new RuntimeException(
+                                "Malformed line in file " + fromFileName + ": " + line);
+                    }
+
+                    String operation = parts[0].trim();
+                    int value = Integer.parseInt(parts[1].trim());
+
+                    if (operation.equals(OPERATION_BUY)) {
+                        totalBuy += value;
+                    } else if (operation.equals(OPERATION_SUPPLY)) {
+                        totalSupply += value;
+                    }
```

This violates the checklist item: "If encountering malformed lines or invalid numbers, handle via exception path consistently with the overall error handling approach (i.e., do not silently ignore)." Currently, when `operation` is neither `supply` nor `buy`, the code simply does nothing (silently skips the line). Instead, consider explicitly treating unexpected operations as malformed input and throw a RuntimeException that includes the `fromFileName` and the raw `line` so it is handled consistently with other malformed/parse errors. For example, after parsing `operation` and `value`, add a validation such as:

```
if (!operation.equals(OPERATION_BUY) && !operation.equals(OPERATION_SUPPLY)) {
    throw new RuntimeException("Malformed line in file " + fromFileName + ": " + line);
}
```

This ensures malformed operation types are not silently ignored and are reported the same way as other parsing errors.